File: Vue.Net/VOL.Entity/DomainModels/Marketing/MktMember.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using VOL.Entity.SystemModels;

namespace VOL.Entity.DomainModels.Marketing
{
    [Table("mkt_members")]
    [Entity(ApiInput = typeof(ApiMktMemberInput), ApiOutput = typeof(ApiMktMemberOutput))]
    public class MktMember : BaseEntity
    {
        [Key]
        [Display(Name = "Id")]
        [Column(TypeName = "bigint")]
        [Required(AllowEmptyStrings = false)]
        public long Id { get; set; }

        [Display(Name = "Username")]
        [Column(TypeName = "nvarchar(100)")]
        public string Username { get; set; }

        [Display(Name = "Nickname")]
        [Column(TypeName = "nvarchar(100)")]
        public string Nickname { get; set; }

        [Display(Name = "Phone")]
        [Column(TypeName = "nvarchar(20)")]
        public string Phone { get; set; }

        [Display(Name = "Email")]
        [Column(TypeName = "nvarchar(200)")]
        public string Email { get; set; }

        [Display(Name = "Avatar")]
        [Column(TypeName = "nvarchar(500)")]
        public string Avatar { get; set; }

        [Display(Name = "LevelId")]
        [Column(TypeName = "bigint")]
        public long? LevelId { get; set; }

        [Display(Name = "ParentId")]
        [Column(TypeName = "bigint")]
        public long? ParentId { get; set; }

        [Display(Name = "Balance")]
        [Column(TypeName = "decimal(18,2)")]
        public decimal Balance { get; set; }

        [Display(Name = "FrozenBalance")]
        [Column(TypeName = "decimal(18,2)")]
        public decimal FrozenBalance { get; set; }

        [Display(Name = "CommissionBalance")]
        [Column(TypeName = "decimal(18,2)")]
        public decimal CommissionBalance { get; set; }

        [Display(Name = "TotalRecharge")]
        [Column(TypeName = "decimal(18,2)")]
        public decimal TotalRecharge { get; set; }

        [Display(Name = "TotalWithdrawal")]
        [Column(TypeName = "decimal(18,2)")]
        public decimal TotalWithdrawal { get; set; }

        [Display(Name = "TotalCommission")]
        [Column(TypeName = "decimal(18,2)")]
        public decimal TotalCommission { get; set; }

        [Display(Name = "Ip")]
        [Column(TypeName = "nvarchar(100)")]
        public string Ip { get; set; }

        [Display(Name = "LastLoginTime")]
        [Column(TypeName = "datetime")]
        public DateTime? LastLoginTime { get; set; }

        [Display(Name = "LastLoginIp")]
        [Column(TypeName = "nvarchar(100)")]
        public string LastLoginIp { get; set; }

        [Display(Name = "Status")]
        [Column(TypeName = "int")]
        public int Status { get; set; }

        [Display(Name = "RealName")]
        [Column(TypeName = "nvarchar(50)")]
        public string RealName { get; set; }

        [Display(Name = "IdCard")]
        [Column(TypeName = "nvarchar(50)")]
        public string IdCard { get; set; }

        [Display(Name = "BankName")]
        [Column(TypeName = "nvarchar(200)")]
        public string BankName { get; set; }

        [Display(Name = "BankAccount")]
        [Column(TypeName = "nvarchar(100)")]
        public string BankAccount { get; set; }

        [Display(Name = "BankBranch")]
        [Column(TypeName = "nvarchar(200)")]
        public string BankBranch { get; set; }

        [Display(Name = "InviteCode")]
        [Column(TypeName = "nvarchar(50)")]
        public string InviteCode { get; set; }

        [Display(Name = "CreateID")]
        [Column(TypeName = "int")]
        public int? CreateID { get; set; }

        [Display(Name = "Creator")]
        [Column(TypeName = "nvarchar(200)")]
        public string Creator { get; set; }

        [Display(Name = "CreateDate")]
        [Column(TypeName = "datetime")]
        public DateTime? CreateDate { get; set; }

        [Display(Name = "ModifyID")]
        [Column(TypeName = "int")]
        public int? ModifyID { get; set; }

        [Display(Name = "Modifier")]
        [Column(TypeName = "nvarchar(200)")]
        public string Modifier { get; set; }

        [Display(Name = "ModifyDate")]
        [Column(TypeName = "datetime")]
        public DateTime? ModifyDate { get; set; }
    }
}

File: Vol.Vue/src/extension/marketing/Mkt_Member.js
export default {
  table: {
    key: 'Id',
    footer: 'Foots',
    cnName: '会员',
    name: 'Mkt_Member',
    url: '/Mkt_Member/',
    sortName: 'CreateDate',
    searchFormFields: { Username: '', Nickname: '', Phone: '', LevelId: '', Status: '' },
    columns: [
      { field: 'Id', title: 'Id', type: 'int', width: 90, hidden: true, readonly: true, require: true, align: 'left' },
      { field: 'Username', title: '用户名', type: 'string', link: true, width: 120, require: true, align: 'left', sortable: true },
      { field: 'Nickname', title: '昵称', type: 'string', width: 120, align: 'left' },
      { field: 'Phone', title: '手机号', type: 'string', width: 130, align: 'left' },
      { field: 'LevelId', title: '等级', type: 'int', bind: { key: 'mkt_level', data: [] }, width: 100, align: 'left' },
      { field: 'Balance', title: '余额', type: 'decimal', width: 110, align: 'left' },
      { field: 'FrozenBalance', title: '冻结余额', type: 'decimal', width: 110, align: 'left' },
      { field: 'CommissionBalance', title: '佣金余额', type: 'decimal', width: 110, align: 'left' },
      { field: 'TotalRecharge', title: '总充值', type: 'decimal', width: 110, align: 'left' },
      { field: 'TotalWithdrawal', title: '总提现', type: 'decimal', width: 110, align: 'left' },
      { field: 'Status', title: '状态', type: 'int', bind: { key: 'member_status', data: [] }, width: 90, require: true, align: 'left' },
      { field: 'Creator', title: '创建人', type: 'string', width: 130, readonly: true, align: 'left' },
      { field: 'CreateDate', title: '创建时间', type: 'datetime', width: 150, readonly: true, align: 'left', sortable: true }
    ],
    detail: {
      cnName: '明细',
      columns: [],
      sortName: 'CreateDate',
      key: 'Id',
      delKeys: []
    }
  },
  extend: {},
  text: '',
  methods: {}
}
